packages/devtools-reps/src/object-inspector/components/ObjectInspector.js

```diff
@@ -130,25 +147,58 @@ class ObjectInspector extends Component<Props> {
     const { loadedProperties } = this.props;
     const { cachedNodes } = this;
 
-    return getChildren({
+    const children = getChildren({
       loadedProperties,
       cachedNodes,
       item
     });
+    if (Array.isArray(children)) {
+      return children.map(i => this.getEvaluatedItem(i));
+    }
+
+    if (children) {
+      return this.getEvaluatedItem(children);
+    }
+
+    return [];
   }
 
   getRoots(): Array<Node> {
     return this.props.roots;
   }
 
+  getEvaluatedItem(item: Node): Node {
```

Comment:
it might be simpler to have an evaluations utils module that has some of this tree node logic...

Reply:
I'm not a 100% convinced here. We could have a util function for that, e.g. `getEvaluatedItem(item: Node, evaluations: Map)` , but the only consumer would eventually be this component.

Am I missing something?

Reply:
moved to node utils